packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -13,6 +13,286 @@ import 'focus_manager.dart';
 import 'focus_scope.dart';
 import 'framework.dart';
 
+/// The factory to construct a `PlatformViewSurface`.
+///
+/// It is used when constructing a `PlatformViewControllerWidget`.
+/// A sample implementation could be:
+/// ```dart
+/// surfaceFactory: (BuildContext context, PlatformViewController controller, int id) {
+///        return PlatformViewSurface(
+///            id: id,
+///            gestureRecognizers: gestureRecognizers,
+///            controller: controller,
+///        );
+///       }
+/// ```
+///
+/// See also:
+/// * `PlatformViewSurface` for more details.
+typedef PlatformViewSurfaceFactory = PlatformViewSurface Function(BuildContext context, PlatformViewController controller, int id);
+
+/// The function to create a `PlatformViewController`.
+///
+/// The implementer of a new platform view is responsible to implement this method when constructing a `PlatformViewControllerWidget`.
+/// This method will get invoked when the state of `PlatformViewControllerWidget` is initialized.
+///
+/// A simple example of implementing this function for foo platform can be:
+/// ```dart
+/// FooPlatformViewController createPlatformView(PlatformViewCreationParams params) {
+///   final FooPlatformViewController controller = FooPlatformViewController(id: params.id);
+///   params.onPlatformViewCreated(id);
+///   return controller;
+/// }
+///```
+typedef CreatePlatformView = PlatformViewController Function(PlatformViewCreationParams params);
+
+/// The widget responsible for handling the life cycle and focus for implementing platform view widget.
+///
+/// To implement a new platform view widget, return this widget in the `build` method.
+/// e.g:
+/// ```dart
+/// class FooPlatformView extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return PlatformViewControllerWidget(
+///       createCallback: createFooWebView,
+///       surfaceFactory: (BuildContext context, PlatformViewController controller, int id) {
+///        return PlatformViewSurface(
+///            id: id,
+///            gestureRecognizers: gestureRecognizers,
+///             controller: controller,
+///        );
+///       }, onPlatformViewCreated: (int id) {
+///         _onPlatformViewCreated(id);
+///      },
+///    );
+///   }
+/// }
+/// ```
+class PlatformViewControllerWidget extends StatefulWidget {
```

Comment:
As discussed we want a different name here, we actually don't want that name to be PlatformView as we have other plans for a widget with that name, and as people are unlikely to be using this so we don't want them to pick this widget when they look for PlatformView in the api docs.

One proposal was PlatformViewLink (it "links" the platform view to the widget layer).